### PR TITLE
Made goal list editable in group creation pipeline

### LINKED ISSUE #65: Make goals in the the goal list editable in the group pipeline.

### PR DESCRIPTION
**Pull request by yashasvi-karri**
# Made goal list editable in group creation pipeline

## Description

Fixes #65 

## Screenshot/Recording (if applicable)

| Screenshot 1 | Screenshot 2 | Screenshot 3 |
| --- | --- | --- |
| Screenshot 4 | Screenshot 5 | Screenshot 6 |

### CODE REVIEW

File: Vineyard/Vineyard/CreateGoalView.swift [hunk 1]
```diff
@@ -8,44 +8,45 @@
 import SwiftUI
 
 struct CreateGoalView: View {
-//    @Binding var goals: [String] // Binding to update the goals list
+    //    @Binding var goals: [String] // Binding to update the goals list
```
Comment (rahulon12) [inline Vineyard/Vineyard/CreateGoalView.swift]: Remove comment

File: Vineyard/Vineyard/CreateGoalView.swift [hunk 2]
```diff
@@ -8,44 +8,45 @@
...
+    @Binding var indexOfGoal: Int
     @Binding var goals: [Resolution]
-       
-    @State private var action: String = "" // the name of the goal
-    @State private var description: String = "" // the name of the goal
-    @State private var quantity: String = ""
+    @Binding var editMode: Bool
+    
+    @State var action: String = "" // the name of the goal
+    @State var description: String = "" // the name of the goal
```
Comment (rahulon12) [inline Vineyard/Vineyard/CreateGoalView.swift]: nit:
```suggestion
    @State var description: String = "" // the description of the goal
```

File: Vineyard/Vineyard/CreateGoalView.swift [hunk 3]
```diff
@@ -8,44 +8,45 @@
...
+    
+    @Binding var indexOfGoal: Int
     @Binding var goals: [Resolution]
-       
-    @State private var action: String = "" // the name of the goal
-    @State private var description: String = "" // the name of the goal
-    @State private var quantity: String = ""
+    @Binding var editMode: Bool
+    
+    @State var action: String = "" // the name of the goal
```
Comment (rahulon12) [inline Vineyard/Vineyard/CreateGoalView.swift]: `action` reminds me of a `Button`s action or something similar, we can rename this:

```suggestion
    @State var goalName: String = "" // the name of the goal
```

File: Vineyard/Vineyard/CreateGroupView.swift [hunk 4]
```diff
@@ -13,12 +13,12 @@ struct CreateGroupView: View {
...
     @Environment(GroupsListViewModel.self) var viewModel
     @Environment(\.dismiss) var dismiss
+    
         
     @State var groupName: String = ""
     @State var resolution: String = ""
     @State var deadline: Date = Date.now
     
-    
 //    @State var isValid: Bool = false
```
Comment (rahulon12) [inline Vineyard/Vineyard/CreateGroupView.swift]: Comments

File: Vineyard/Vineyard/CreateGroupView.swift [hunk 5]
```diff
@@ -103,14 +103,20 @@ struct TextFieldTitleView: View {
 
 struct GoalsListView: View {
     @Environment(GroupsListViewModel.self) var viewModel
-    @State private var goals : [Resolution] = []
+    @State var goals : [Resolution] = []
+    @State var indexOfGoal: Int = -1
+//    @State var goal: Resolution
```
Comment (rahulon12) [inline Vineyard/Vineyard/CreateGroupView.swift]: Comments

**Review by rahulon12 — COMMENTED** [review]
A few comments

**Review by sankaet8 — APPROVED** [review]
lgtm just one quick question, do we want the three line thing 
Screenshot 2024-11-08 at 11 33 39
or something else that indicates editing? Or is the three line thing for something else?

**Comment by sankaet8** [comment]
> lgtm just one quick question, do we want the three line thing Screenshot 2024-11-08 at 11 33 39 or something else that indicates editing? Or is the three line thing for something else?

resolved

**Pull request merged**